src/include/api.h

```diff
@@ -32,24 +32,31 @@
 #define WT_SINGLE_THREAD_CHECK_STOP(s)
 #endif
 
+#define API_SESSION_PUSH(s, h, n, dh)       \
+    WT_DATA_HANDLE *__olddh = (s)->dhandle; \
+    const char *__oldname = (s)->name;      \
+    (s)->dhandle = (dh);                    \
+    (s)->name = (s)->lastop = #h "." #n
+#define API_SESSION_POP(s)  \
+    (s)->dhandle = __olddh; \
+    (s)->name = __oldname
+
 /* Standard entry points to the API: declares/initializes local variables. */
-#define API_SESSION_INIT(s, h, n, dh)                               \
-    WT_TRACK_OP_DECL;                                               \
-    WT_DATA_HANDLE *__olddh = (s)->dhandle;                         \
-    const char *__oldname = (s)->name;                              \
-    (s)->dhandle = (dh);                                            \
-    (s)->name = (s)->lastop = #h "." #n;                            \
-    /*                                                              \
-     * No code before this line, otherwise error handling  won't be \
-     * correct.                                                     \
-     */                                                             \
-    WT_TRACK_OP_INIT(s);                                            \
-    WT_SINGLE_THREAD_CHECK_START(s);                                \
-    __wt_op_timer_start(s);                                         \
-    WT_ERR(WT_SESSION_CHECK_PANIC(s));                              \
-    /* Reset wait time if this isn't an API reentry. */             \
-    if (__oldname == NULL)                                          \
-        (s)->cache_wait_us = 0;                                     \
+#define API_SESSION_INIT(s, h, n, dh)                              \
+    WT_TRACK_OP_DECL;                                              \
+    API_SESSION_PUSH(s, h, n, dh);                                 \
+    /*                                                             \
+     * No code before this line, otherwise error handling won't be \
+     * correct.                                                    \
+     */                                                            \
+    WT_ERR(WT_SESSION_CHECK_PANIC(s));                             \
+    WT_SINGLE_THREAD_CHECK_START(s);                               \
+    WT_ERR(__wt_txn_err_chk(s));                                   \
+    WT_TRACK_OP_INIT(s);                                           \
```

Comment:
Did you deliberately move this one further down? It used to be before the single thread check.

Reply:
@agorrod 
> Did you deliberately move this one further down? It used to be before the single thread check.

Yes -- I put all the error/panic checks first, and then initialization afterward, they were randomly mixed up before. There's no correctness issue, it just seemed cleaner to me.